tests/protocol.cc

```diff
@@ -163,7 +164,7 @@ static test_return_t GEARMAN_COMMAND_ECHO_REQ_overrun_TEST(void *)
 
   gearman_connection_st *connection1;
   ASSERT_TRUE(connection1= gearman_connection_create(universal, GEARMAN_DEFAULT_TCP_HOST, libtest::default_port()));
-  ASSERT_TRUE(connection1);
+  ASSERT_TRUE(typeid(decltype(*connection1)).name()==typeid(gearman_connection_st).name());
```

Comment:
> "Assert that connection1 and gearman_connection_st have the same type".

absolute exactly right. Because simple `ASSERT_TRUE` is not accurate enough.